### PR TITLE
Fake queue stats when there is padding

### PR DESCRIPTION
**Pull request by algesten**

### CODE REVIEW

File: src/streams/send.rs [hunk 1]
```diff
@@ -680,6 +681,28 @@ impl StreamTx {
...
+            } else {
+                QueuePriority::Media
+            };
+
+            let fake_packets = self.padding / AVERAGE_PADDING_PACKET_SIZE;
+
+            snapshot.size += self.padding;
+            snapshot.packet_count += fake_packets as u32;
+
+            // TODO: Should this be faked in relation to size of padding?
```
Comment (k0nserv) [inline src/streams/send.rs:702]: My removal of the VecDeque we used for `self.padding` probably doesn't help here. If we still had that we could keep a `queued_at: Instant` field with every padding request and produce a better value here.

To answer the TODO: Shouldn't it be scaled with the number of packets?

File: src/streams/send.rs [hunk 2]
```diff
@@ -680,6 +681,28 @@ impl StreamTx {
         //    first ever poll_packet().
         let use_for_padding = self.is_audio.map(|i| !i).unwrap_or(false);
 
+        let mut snapshot = self.send_queue.snapshot(now);
+
+        // Fake some stats for padding.
+        if self.padding > 0 {
+            // TODO: Be more scientific about this factor.
+            const AVERAGE_PADDING_PACKET_SIZE: usize = 200;
```
Comment (k0nserv) [inline src/streams/send.rs:689]: At the very least I think we can make this higher. Even at 0.5Mbit/s the padding burst will be ~3kb and as such we'll probably use a lot of fairly large packets for spurious resends. 700-900 seems like a better first guess than 200

File: src/streams/send.rs (hunk 1 above)
Comment (algesten) [reply src/streams/send.rs:702]: Yeah. I was thinking that too. However. Maybe it's an all-or-nothing situation. In the sense that either we make perfect stats (like we did in the past), or we get away with not perfect stats. I do really like the simplification you did.

File: src/streams/send.rs (hunk 2 above)
Comment (algesten) [reply src/streams/send.rs:689]: Ok. I change that.

File: src/streams/send.rs (hunk 1 above)
Comment (k0nserv) [reply src/streams/send.rs:702]: Yea, this is still an improvement. If we just scale it with the number of packets maybe that'd be okay for now

**Review by k0nserv — APPROVED** [review]

**Pull request merged**